precompiles/prototype/prototype_test.go

```diff
@@ -0,0 +1,339 @@
+package prototype
+
+import (
+	"encoding/json"
+	"testing"
+
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+	ethermint "github.com/zeta-chain/ethermint/types"
+	"github.com/zeta-chain/zetacore/precompiles/types"
+	"github.com/zeta-chain/zetacore/testutil/keeper"
+)
+
+func Test_IPrototypeContract(t *testing.T) {
+	/*
+		Configuration
+	*/
+
+	var encoding ethermint.EncodingConfig
+	appCodec := encoding.Codec
+	k, _, _, _ := keeper.FungibleKeeper(t)
+	gasConfig := storetypes.TransientGasConfig()
+
+	/*
+		Contract and ABI tests
```

Comment:
Consider breaking up the function into multiple tests using `t.Run(testname)`

It might be easier to debug when tests break .